src/Microsoft.Restier.Core/ApiConfiguration.cs

```diff
@@ -65,5 +68,46 @@ public IServiceProvider ServiceProvider
         {
             return this.serviceProvider.GetService<T>();
         }
+
+        internal TaskCompletionSource<IEdmModel> CompeteModelGeneration(out Task<IEdmModel> running)
+        {
+            var source = new TaskCompletionSource<IEdmModel>(TaskCreationOptions.AttachedToParent);
+            var exist = Interlocked.CompareExchange(ref modelTask, source.Task, null);
```

Comment:
Shall we name it `runningTask` to make the code clearer? For example, if we wrote `if (runningTask != null)`, we knew that there is no ongoing task.